swift_scality_backend/diskfile.py

```diff
@@ -162,28 +172,30 @@ def _do_http(self, caller_name, handlers, method, path, headers=None):
         address, port = self.sproxyd_hosts.next()
         safe_path = self.base_path + urllib.quote(path)
 
-        conn = None
-
         def unexpected_http_status(response):
-            message = response.read()
+            message = response.data
 
             raise SproxydHTTPException(
                 '%s: %s' % (caller_name, message),
                 ipaddr=address, port=port,
-                path=self.base_path,
+                path=safe_path,
                 http_status=response.status,
                 http_reason=response.reason)
 
-        with swift.common.exceptions.ConnectionTimeout(self.conn_timeout):
-            conn = swift.common.bufferedhttp.http_connect_raw(
-                address, port, method, safe_path, headers)
+        pool = self.http_pools.connection_from_host(address, port)
+        response = None
+        try:
+            response = pool.request(method, safe_path, headers=headers, preload_content=False)
+            handler = handlers.get(response.status, unexpected_http_status)
 
-        with contextlib.closing(conn), eventlet.Timeout(self.proxy_timeout):
-            resp = conn.getresponse()
-            status = resp.status
+            self.logger.debug('The HTTP connection pool to %s:%d serviced %d '
+                              'requests. Its max size ever is %d.', pool.host,
+                              pool.port, pool.num_requests, pool.num_connections)
 
-            handler = handlers.get(status, unexpected_http_status)
-            return handler(resp)
+            return handler(response)
+        finally:
+            if response:
+                response.release_conn()
```

Comment:
No, only if running `handler` didn't fail, otherwise some data could still be on the wire?